Parquet/src/main/java/io/deephaven/parquet/ColumnWriterImpl.java

```diff
@@ -95,27 +101,22 @@ private void initWriter() {
     }
 
     @Override
-    public void addDictionaryPage(Object dictionaryValues, int valuesCount) throws IOException {
+    public void addDictionaryPage(final Object dictionaryValues, final int valuesCount) throws IOException {
         if (pageCount > 0) {
             throw new RuntimeException("Attempting to add dictionary past the first page");
         }
         BulkWriter dictionaryWriter = getWriter(column.getPrimitiveType());
         //noinspection unchecked
         dictionaryWriter.writeBulk(dictionaryValues, valuesCount);
         dictionaryOffset = writeChannel.position();
-        if (dictionaryOffset == 0) {//Ugly hack to deal with parquet readers that assume dictionary is not there if the offset is 0
```

Comment:
I assume parquet-tools and/or drill don't mind this change?

Reply:
Parquet tool doesn't, trying drill would be... complicated.  I could try pyarrow.

Reply:
Just tried pyarrow, it worked:

```
cfs@erke 09:44:22 ~/py
$ python3
Python 3.8.10 (default, Jun  2 2021, 10:49:15) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas as pd
>>> import pyarrow.parquet as pq
>>> t = pq.read_table('/a0/h/cfs/dh/oss3/io.deephaven.db.v2.parquet.ParquetTableReadWriteTest_root/ParquetTest_emptyFlatParquet_test.parquet')
>>> t.to_pandas()
Empty DataFrame
Columns: [someStringColumn, nonNullString, nonNullPolyString, someIntColumn, someLongColumn, someDoubleColumn, someFloatColumn, someBoolColumn, someShortColumn, someByteColumn, someCharColumn, someTime, someKey, nullKey, someSerializable]
Index: []

```